installer.sh

```diff
@@ -49,22 +38,64 @@ echo ""
 echo "Testing if the script works"
 echo ""
 
-timeout 2.5s ./venv/bin/python ./main.py
+# Run the Python script in the background
+./venv/bin/python ./main.py &
+# Get the background process's PID
+PYTHON_PID=$!
+# Sleep for 2.5 seconds
+sleep 2.5
+# Kill the background process
+kill $PYTHON_PID
 
 echo "Test might've worked, did it spit out any errors?"
 echo "Commands executed."
 
-# Turn steam-presence/bin/python into steam-presence/venv/bin/python and replace "/home/deck" with current directory in steam-presence.service file
-echo "Setting up service file"
-mkdir -p "$HOME/.config/systemd/user"
-sed -e "s~steam-presence/bin/python~steam-presence/venv/bin/python~g" -e "s~/home/deck/steam-presence~$PWD~g" "$PWD/steam-presence.service" > $HOME/.config/systemd/user/steam-presence.service
-
+# Create a shell script wrapper for macOS
+if [[ "$OSTYPE" == "darwin"* ]]; then
+  echo '#!/bin/bash' > steam-presence
+  echo "DIR=\"\$( cd \"\$( dirname \"\${BASH_SOURCE[0]}\" )\" && pwd )\"" >> steam-presence
+  echo "exec \"\$DIR/venv/bin/python\" \"\$DIR/main.py\"" >> steam-presence
+  chmod +x steam-presence
+fi
 
-echo "Starting service"
-systemctl --user daemon-reload
-systemctl --user --now enable "steam-presence.service"
+# Detect OS and perform OS-specific tasks
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+  echo "Setting up service file for Linux"
+  mkdir -p "$HOME/.config/systemd/user"
+  sed -e "s~steam-presence/bin/python~steam-presence/venv/bin/python~g" -e "s~/home/deck/steam-presence~$PWD~g" "$PWD/steam-presence.service" > $HOME/.config/systemd/user/steam-presence.service
+  echo "Starting service"
+  systemctl --user daemon-reload
+  systemctl --user --now enable "steam-presence.service"
+elif [[ "$OSTYPE" == "darwin"* ]]; then
+  echo "Setting up launchd plist file for macOS"
+  PLIST="$HOME/Library/LaunchAgents/com.steampresence.app.plist"
```

Comment:
Naming could be a bit better, for example "com" implies commercial, if you want to use that naming scheme then i would suggest "com.github.justtemmie.steam-presence.plist" because it is a more accurate description.

Reply:
I switch to com.github.justtemmie.steam-presence.plist. 'com' is the convention, although there appears to not be any hard rule for that. What about changing to 'dev' or something else? They are expecting domain-style naming.

Reply:
Naming it com.github.<author>.<project> is pretty normal in flatpaks if you know anything about those so i think that's fine